vector3/vector3.go

```diff
@@ -241,11 +242,11 @@ func (v Vector[T]) Format(format string) string {
 }
 
 func (v Vector[T]) MinComponent() T {
-	return T(math.Min(float64(v.x), math.Min(float64(v.y), float64(v.z))))
+	return min(v.x, min(v.y, v.z))
```

Comment:
I think it's valid to put it all into one call
https://tip.golang.org/ref/spec#Min_and_max

```suggestion
	return min(v.x, v.y, v.z)
```